zebra-rpc/src/methods.rs

```diff
@@ -994,6 +992,19 @@ where
     }
 }
 
+/// Returns the best chain tip height of `latest_chain_tip`,
+/// or an RPC error if there are no blocks in the state.
+pub fn best_chain_tip_height<Tip>(latest_chain_tip: &Tip) -> Result<Height>
+where
+    Tip: ChainTip + Clone + Send + Sync + 'static,
+{
+    latest_chain_tip.best_tip_height().ok_or(Error {
```

Comment:
On this note it might be nice to have a `make_server_error` fn in the future.

Reply:
Something like this, but maybe with an `error_code: Into<Option<i64>>` argument.

```rust
impl<T, E> MapServerError for Result<T, E> where E: Debug {
    fn map_server_error(result: Self) -> Result<T, jsonrpc_core::Error> {
        result.map_err(|error| jsonrpc_core::Error {
            code: ErrorCode::ServerError(0),
            message: format!("{error:?}"),
            data: None,
        })
    }
}

impl<T> OkOrServerError for Option<T> {
    fn ok_or_server_error<S: ToString>(option: Self, message: S) -> Result<T, jsonrpc_core::Error> {
        option.ok_or(jsonrpc_core::Error {
            code: ErrorCode::ServerError(0),
            message: message.to_string(),
            data: None,
        })
    }
}
```